x/emissions/keeper/inference_synthesis/synth_palette_weight.go

```diff
@@ -47,25 +45,25 @@ func (p *SynthPalette) CalcWeightsGivenWorkers() (RegretInformedWeights, error)
 	// Normalize the regrets and find the max normalized regret among them
 	normalizedInfererRegrets := make(map[Worker]Regret)
 	maxRegret := alloraMath.ZeroDec()
-	for i, worker := range p.Inferers {
-		regretFrac, err := p.InfererRegrets[worker].regret.Quo(stdDevRegretsPlusFTolerance)
+	for address, worker := range p.InfererRegrets {
+		regretFrac, err := worker.regret.Quo(stdDevRegretsPlusFTolerance)
 		if err != nil {
 			return RegretInformedWeights{}, errorsmod.Wrapf(err, "Error calculating regret fraction")
 		}
-		normalizedInfererRegrets[worker] = regretFrac
-		if i == 0 || regretFrac.Gt(maxRegret) {
+		normalizedInfererRegrets[address] = regretFrac
+		if regretFrac.Gt(maxRegret) {
```

Comment:
What about the case where the regret is less than zero? We still need to initialize the max regret on the first round

Reply:
+1 what @br4e said...
what ab negative regrets? what ab initialization via first value?